edgeless_telemetry/src/elasticsearch_api.rs

```diff
@@ -0,0 +1,203 @@
+use chrono::{DateTime, Utc};
+use elasticsearch::indices::IndicesCreateParts;
+use elasticsearch::{
+    auth::Credentials, http::transport::SingleNodeConnectionPool, http::transport::TransportBuilder, Elasticsearch, IndexParts, SearchParts,
+};
+use serde_json::json;
+use serde_json::Value;
+use url::Url;
+
+//static mut variable to hold the counter
+static mut COUNTER: u32 = 0;
+
+/// Establish connection to specified ES endpoint.
+/// # Returns
+/// A Result indicating success or failure.
+pub fn es_create_client() -> Result<Elasticsearch, Box<dyn std::error::Error>> {
+    //define ES endpoint configs
+
+    let url = Url::parse("elastic_endpoint_url")?; //contant Panagiotis Antoniou(Aegis) for details
+    let credentials = Credentials::Basic("username".into(), "password".into()); //contact Panagiotis Antoniou(Aegis) for details
+    let conn_pool = SingleNodeConnectionPool::new(url);
+    let transport = TransportBuilder::new(conn_pool).auth(credentials).build()?;
+
+    Ok(Elasticsearch::new(transport))
+}
+
+/// Creates an Elasticsearch index with a specified mapping.
+/// # Arguments
+/// * `client_result` - the Elasticsearch client or an error.
+/// # Returns
+/// A Result indicating success or failure.
+//To be called once on init.
+pub async fn es_create_index(
+    client_result: &Result<Elasticsearch, Box<dyn std::error::Error>>,
```

Comment:
You probably don't want to pass a result here, but extract the value in the function calling this and only call it with a client.

Reply:
In fact, you probably want the client to be a wrapper struct that is created with a new method and has methods to interact with ES.